DSharpPlus/DiscordClient.cs

```diff
@@ -1555,18 +1555,18 @@ internal static DiscordPresence InternalGetUserPresence(ulong userid)
             };
         }
 
-        internal static int InternalAddPermission(int before, Permission p)
+        internal static Permission InternalAddPermission(Permission before, Permission p)
         {
-            int after = before;
-            after &= (int)p;
+            Permission after = before;
+            after &= p;
             Console.WriteLine(before + " " + after);
             return after;
         }
 
-        internal static int InternalRemovePermission(int before, Permission p)
+        internal static Permission InternalRemovePermission(Permission before, Permission p)
         {
-            int after = before;
-            after |= (int)p;
```

Comment:
This is the logic error that prevented permission setting from working correctly.  Instead of | (setting the bit in `after` if it's set in `p`), you want to essentially do:
```
after &= ( p ^ 0xffffffffffffffff );
```
Because the ~ operator (complement) is basically a shorthand for that (xor the bit you want to clear with an all-ones variable of the same length), the final form works out to:
```
after &= ~p;
```